packages/zendesk-adapter/src/user_utils.ts

```diff
@@ -46,6 +56,157 @@ const areUsers = (values: unknown): values is User[] => {
   return true
 }
 
+const replaceConditionsAndActionsCreator = (
+  params: UserFieldsParams[],
+  isIdNumber = false,
+): UserReplacer => (instance, mapping) => (
+  params.flatMap(replacerParams => {
+    const conditions = _.get(instance.value, replacerParams.fieldName)
+    const { typeName } = instance.elemID
+    // Coditions can be undefined - in that case, we don't want to log a warning
+    if (conditions === undefined
+      || !isCorrectConditions(conditions, typeName)) {
+      return []
+    }
+    return conditions
+      .flatMap((condition, i) => {
+        const fieldNamesToReplace = replacerParams.fieldsToReplace.map(f => f.name)
+        const conditionValue = conditionFieldValue(condition, typeName)
+        if (!fieldNamesToReplace.includes(conditionValue)) {
+          return []
+        }
+        const valueRelativePath = replacerParams.fieldsToReplace
+          .find(f => f.name === conditionValue)?.valuePath ?? ['value']
+        const value = _.get(condition, valueRelativePath)?.toString()
+        if (value !== undefined) {
```

Comment:
maybe it will be less confusing if we do like this. what do you think? 
```suggestion
        if (value === undefined) {
        return []
        }
```